### DISCUSSION TITLE
how to get the dimensions of constraints

### DISCUSSION THREAD

**Discussion by P11GH** [yalmip/YALMIP#1253] [Problems using YALMIP] [answered]
I am working on an problem with constraints like
constr=[A x - b == 0; C x - d <= 0],
where the dimentions of A, b, C, d, and x might change depending on applications. I want to get their dimensions before solving them. Does anyone know how to do it?

For example, I get the following information on the constraints. Is there any way to obtain their dimension 24 x 1？
<img width="476" alt="image" src="https://user-images.githubusercontent.com/50893243/228142867-3ba23449-6fc8-46f2-b926-ecee3dc13c09.png">

**Comment by johanlofberg** [comment, marked as answer]
`size(sdpvar(constr(i)))`

This assumes there is no high-level modelling in the model which will lead to additional constraints and variables in the model actually sent to the solver

FYI, the huge coefficients you have in your constraints do note bode well for the numerics in the solver

**Reply by P11GH** [reply]
thank you very much. The huge coefficients are caused by the big M

**Reply by johanlofberg** [reply]
Ok, you should try to put effort on coming up with better M, or a better model in general since those can cause absolute havoc. Big-M models should be called "Small-but-sufficiently-large-M".

**Reply by P11GH** [reply]
thank you very much
